src/main/java/coydir/storage/JsonAdaptedPerson.java

```diff
@@ -149,24 +154,23 @@ public Person toModelType() throws IllegalValueException {
 
         if (department == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
-                        Department.class.getSimpleName()));
+                    Department.class.getSimpleName()));
         }
         if (!Department.isValidDepartment(department)) {
             throw new IllegalValueException(Department.MESSAGE_CONSTRAINTS);
         }
         final Department modelDepartment = new Department(department);
 
+        final Address modelAddress;
         if (address == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Address.class.getSimpleName()));
-        }
-        if (!Address.isValidAddress(address)) {
-            throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
+            modelAddress = new Address();
+        } else {
+            if (!Address.isValidAddress(address)) {
+                throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
+            }
+            modelAddress = new Address(address);
         }
-        final Address modelAddress = new Address(address);
 
-        if (leave == null) {
```

Comment:
Why remove this part?

Reply:
Leave will never be null actually so its redundant

Reply:
This class (`JsonAdaptedPerson`) is meant for reading and writing JSON files, so it's not so simple. We shouldn't assume that the JSON files will be untampered.

![image](https://user-images.githubusercontent.com/78629832/198008184-0df4242e-cea4-4d2c-8997-ca1ac03f2081.png)

If say the "leave" portion is changed to null, we can indeed get an error. So in line with defensive programming, it would be better to retain this.

Reply:
Good point so we need to keep the null case for all of the fields?

Reply:
Yeah better to.

Reply:
Added the Null Checks